core/arch/arm/kernel/generic_boot.c

```diff
@@ -821,6 +825,30 @@ static void reset_dt_references(void)
 {
 }
 
+static void carveout_memory_range(void)
+{
+	const struct core_mmu_phys_mem *mem;
+	size_t nelems = 0;
+
+	/* check if nsec ddr had been registered by register_nsec_ddr()*/
+	if (&__start_phys_nsec_ddr_section != &__end_phys_nsec_ddr_section)
+		return;
+
+	nelems = (&__end_phys_ddr_overall_section -
+		&__start_phys_ddr_overall_section) / sizeof(*mem);
+	if (!nelems)
+		return;
+
+	mem = calloc(nelems, sizeof(*mem));
+	if (!mem)
+		panic();
+
+	memcpy((void *)mem, &__start_phys_ddr_overall_section,
+		sizeof(*mem) * nelems);
```

Comment:
```
	memcpy(mem, &__start_phys_ddr_overall_section, sizeof(*mem) * nelems);
``

Reply:
Yes, the "mem" point to a "calloc" address which should be defined without "const".  And (void *) can be removed.